### PR TITLE
Updated bool to struct in uniqueue.go

### LINKED ISSUE #3019: [blocksync] use `struct{}` instead of `bool` in hashset
**What would you like to be added**:
https://github.com/iotexproject/iotex-core/blob/master/blocksync/uniqueue.go
`map[hash.Hash256]bool{}` could be replaced by `map[hash.Hash256]struct{}`
**Why is this needed**:

### PR DESCRIPTION
**Pull request by pramod74**
fix #3019

### CODE REVIEW

File: blocksync/uniqueue.go [hunk 1]
```diff
@@ -13,13 +13,13 @@ import (
 // uniQueue is not threadsafe
 type uniQueue struct {
 	blocks []*peerBlock
-	hashes map[hash.Hash256]bool
+	hashes map[hash.Hash256]struct
```
Comment (Liuhaai) [inline blocksync/uniqueue.go:16]: use `struct{}` here

**Review by Liuhaai — APPROVED** [review]

**Comment by pramod74** [comment]
Thanks!

File: blocksync/uniqueue.go [hunk 2]
```diff
@@ -13,13 +13,13 @@ import (
...
 	blocks []*peerBlock
-	hashes map[hash.Hash256]bool
+	hashes map[hash.Hash256]struct{}
 }
 
 func newUniQueue() *uniQueue {
 	return &uniQueue{
 		blocks: []*peerBlock{},
-		hashes: map[hash.Hash256]bool{},
+		hashes: map[hash.Hash256]struct{},
```
Comment (Liuhaai) [inline blocksync/uniqueue.go:22]: map[hash.Hash256]struct{}{}

**Comment by Liuhaai** [comment]
@pramod74 Could you help fix the error in https://app.circleci.com/pipelines/github/iotexproject/iotex-core/3771/workflows/b868f8d2-73b4-451b-b1f2-a95ec884474d/jobs/15663

**Comment by pramod74** [comment]
Sorry @Liuhaai. I did look into this but have not been able to figure out a fix for the issue. Guidance from someone experienced in programming using Go would help.

**Review by CoderZhi — APPROVED** [review]

**Pull request merged**